frontend/src/pages/breadcrumbs.tsx

```diff
@@ -0,0 +1,40 @@
+import BreadcrumbGroup from '@cloudscape-design/components/breadcrumb-group'
+import {BreadcrumbGroupProps} from '@cloudscape-design/components/breadcrumb-group/interfaces'
+import {CancelableEventHandler} from '@cloudscape-design/components/internal/events'
+
+type Page = 'clusters' | 'custom-images' | 'official-images' | 'users'
+
+const pageBreadcrumbItems: Record<Page, BreadcrumbGroupProps.Item> = {
```

Comment:
suggestion(non-blocking): replicating the same routing structure here for breadcrumbs it's not the best thing to do IMHO,  and i t has to be in sync with changes on existing and new routes.
What you can do here is to take the `path` from `useLocation`, and use it to build a default breadcrumb inside the `Layout` component, which can be overridden with a prop (for example in the `Clusters` case).
If you follow this approach there's no need to enumerate the routes here, and you get a breadcrumb automatically as soon as a new page is added via the `Layout` component.
Example:
```
const location = useLocation()
const breadCrumbText = _.capitalize(location.pathname.split('/')?.[1].replace('-', ' '))
```

Reply:
I like the configuration approach a little bit better because:
- we should make sure we don't just reflect navigations, but the information architecture ([see here](https://cloudscape.design/components/breadcrumb-group/?tabId=usage)). Not a big deal in our case
- we should still i18n-ize the strings (which we can do with your approach as well, just less idal)

My suggestion (see other comments) would be to just associate every page with a key and let the config do the rest, but I'll leave it up to you guys as I don't see strong arguments for either approach

Reply:
+1 for the information architecture, thank you! Resolving the comment